### PR TITLE
PCSX2: Fix stall on branch in delay slot

### PR DESCRIPTION
**Pull request by lightningterror**
Continuation from https://github.com/PCSX2/pcsx2/pull/1783

pcsx2: Fix stall on branch .. in delay slot.
Fixes stall when loading a stage in WRC 3.

Gamedb: Remove stall delay branch patches from WRC3, WRC Rally Evolved/WRC avec Sebastien Loeb.
Bonus, add XgKickHack for the French version of WRC Rally.

### CODE REVIEW

File: pcsx2/x86/ix86-32/iR5900-32.cpp [hunk 1]
```diff
@@ -1311,22 +1311,33 @@ void recompileNextInstruction(int delayslot)
...
 			case 2: case 3: case 4: case 5: case 6: case 7: case 0x14: case 0x15: case 0x16: case 0x17:
-				Console.Warning("branch %x in delay slot!", cpuRegs.code);
-				_clearNeededX86regs();
-				_clearNeededXMMregs();
-				return;
+				check_branch_delay = true;
+		}
+		// Check for branch in delay slot, new code by FlatOut.
+		// Gregory tested this in 2017 using the ps2autotests suite and remarked "So far we return 1 (even with this PR), and the HW 2.
+		// Original PR and discussion at https://github.com/PCSX2/pcsx2/pull/1783 so we don't forget this information.
```
Comment (lightningterror) [inline pcsx2/x86/ix86-32/iR5900-32.cpp]: @ramapcsx2 I believe this should suffice? ;)

**Comment by MrCK1** [comment]
Still requires extensive testing IMO.

**Comment by atomic83GitHub** [comment]
@lightningterror can you remove patches for "WRC avec Sébastien loeb" and add the Xgkick hack? (I forgot to add it).

File: pcsx2/x86/ix86-32/iR5900-32.cpp (hunk 1 above)
Comment (ramapcsx2) [reply pcsx2/x86/ix86-32/iR5900-32.cpp]: Perfect, thanks! :)

**Pull request merged**